doc/api/url.md

```diff
@@ -250,7 +250,448 @@ properties of URL objects:
+property will be percent-encoded. Note that the selection of which characters to
+percent-encode may vary somewhat from what the [`url.parse()`][] and
+[`url.format()`][] methods will produce.
+
+#### url.searchParams
+
+Gets a [`URLSearchParams`][#url_class_urlsearchparams] object representing the
+querystring parameters of the URL.
+
+#### url.username
+
+Gets and sets the username portion of the URL.
+
+```js
+const myURL = new URL('http://abc:xyz@example.com');
+console.log(myURL.username);
+  // Prints abc
+
+myURL.username = '123';
+console.log(myURL.href);
+  // Prints http://123:xyz@example.com
+```
+
+Using the `delete` keyword (e.g. `delete myURL.username`) has no effect but will
+still return `true`.
+
+Any invalid URL characters appearing in the value assigned the `username`
+property will be percent-encoded. Note that the selection of which characters to
+percent-encode may vary somewhat from what the [`url.parse()`][] and
+[`url.format()`][] methods will produce.
+
+#### url.toString([unicode])
+
+* `unicode` {Boolean} If `true`, Unicode characters within the hostname will
+  be encoded as-is without punycode encoding. Defaults to `false`.
+
+The `toString()` method on the `URL` object returns the serialized URL. When
+`unicode` is `false`, the value returned is equivalent to that of `url.href`.
+
+### Class: URLSearchParams
+
+The `URLSearchParams` object provides read and write access to the querystring
+of a `URL`.
+
+```js
+const myURL = new URL('http://example.org/?abc=123');
+console.log(myURL.searchParams.get('abc'));
+  // Prints 123
+
+myURL.searchParams.append('abc', 'xyz');
+console.log(myURL.href);
+  // Prints http://example.org/?abc=123&abc=xyz
+
+myURL.searchParams.delete('abc');
+myURL.searchParams.set('a', 'b');
+console.log(myURL.href);
+  // Prints http://example.org/?a=b
+```
+
+#### Constructor: new URLSearchParams([init])
+
+* `init` {String} The URL querystring
+
+#### urlSearchParams.append(name, value)
+
+* `name` {String}
+* `value` {String}
+* Returns `undefined`
+
+Append a new name-value pair to the query string.
+
+#### urlSearchParams.delete(name)
+
+* `name` {String}
+* Returns `undefined`
+
+Remove all name-value pairs whose name is `name`.
+
+#### urlSearchParams.get(name)
+
+* `name` {String}
+* Returns a string or `undefined`
+
+Returns the value of the first name-value pair whose name is `name`.
+
+#### urlSearchParams.getAll(name)
+
+* `name` {String}
+* Returns a JavaScript Array
+
+Returns the values of all name-value pairs whose name is `name`.
+
+#### urlSearchParams.has(name)
+
+* `name` {String}
+* Returns `true` or `false`
+
+Returns `true` if there is at least one name-value pair whose name is `name`.
+
+#### urlSearchParams.set(name, value)
+
+* `name` {String}
+* `value` {String}
+* Returns `undefined`
+
+Remove any existing name-value pairs whose name is `name` and append a new
+name-value pair.
+
+#### urlSearchParams[Symbol.iterator]()
```

Comment:
Perhaps the `@@iterator` notation used in ES spec would be more concise? Or is this notation already used in the documentation?